sqllogictest/src/parser.rs

```diff
@@ -638,11 +758,32 @@ fn parse_inner<T: ColumnType>(loc: &Location, script: &str) -> Result<Vec<Record
                 // Lines following the "----" are expected results of the query, one value per line.
                 let mut expected_results = vec![];
                 if has_result {
-                    for (_, line) in &mut lines {
-                        if line.is_empty() {
-                            break;
+                    if let Some(e) = &expected_error {
+                        if e.is_empty_inline() {
```

Comment:
This smells a little to me, as it can't be a multiline error here. 🤪

Reply:
Added an "empty" variant to make it clear. 😄